### PR TITLE
Fix issue where response was dropped if the server returns a `String` message id

### PR DESCRIPTION
**Pull request by bkonyi**

### CODE REVIEW

**Comment by natebosch** [comment]
This is a bug on the other side of the communication here. What implementation are we working around bugs in?

I wonder if we should wrap this in with the strict protocol check stuff?

**Comment by bkonyi** [comment]
I don't think so. The [protocol](https://www.jsonrpc.org/specification#request_object) specifies that the ID can be an integer, string, or null. While it's true that the ID will always be an integer for requests made by this package, there's nothing requiring the client to return the ID as an integer. This is just to make things more robust since I ran into this with `package:vm_service`.

**Comment by natebosch** [comment]
> This is just to make things more robust since I ran into this with `package:vm_service`.

That implies a bug in `package:vm_service`.

The protocol allows integers or strings, but it also specifies:

> The Server MUST reply with the same value in the Response object if included.

`'1'` and `1` are not the same value, and it would be allowed for a client to send both those values as IDs and expect for them to be returned exactly so they can be treated as different messages.

The code here assumes `int` because the value we choose as an ID is entirely in our control, and we always choose to send ints.

**Comment by bkonyi** [comment]
Okay, fair enough. We can either close this or allow for this when the strict protocol checks aren't enforced, I'm fine with either.

**Comment by natebosch** [comment]
Sorry I never got back to this PR.

After sitting on it for a bit I think it might be a good idea to move forward with this if you are still interested. I can't think of a reason this change would cause any problems, and it follows the idea of being liberal in what you accept and strict in what you send. We know there isn't a risk of confusing `'1'` and `1` because we only ever send `1`...

Should we go ahead and merge?

**Comment by bkonyi** [comment]
No worries!

I have no objection moving forward with this (although I have fixed the issue upstream in `package:vm_service`... :-)). Like you said, it's probably better to handle this case given we know what type the ID should be rather than silently dropping a response when communicating with a server that doesn't exactly conform with the spec.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 2.2.3-dev (unreleased)
+
+* Fix issue where response was ignored when server returned a `String`
+  message id.
```
Comment (natebosch) [inline CHANGELOG.md:4]:
```suggestion
* Accept responses even if the server converts the ID to a String.
```

**Comment by natebosch** [comment]
Thanks!

**Pull request merged**